### PR TITLE
spirv-val: Emit an error when an OpSwitch target is not an OpLabel

### LINKED ISSUE #1628: OpSwitch error should reference label requirement
The OpSwitch in this example is bad.  It uses a the uint value %7 as a case label, but that's not valid.

```
               OpCapability Shader
               OpMemoryModel Logical GLSL450
               OpEntryPoint GLCompute %1 "foo"
       %uint = OpTypeInt 32 0
     %uint_0 = OpConstant %uint 0
       %void = OpTypeVoid
          %5 = OpTypeFunction %void
          %1 = OpFunction %void None %5
          %6 = OpLabel
          %7 = OpCopyObject %uint %uint_0
               OpSelectionMerge %8 None
               OpSwitch %uint_0 %8 0 %7
          %8 = OpLabel
               OpReturn
               OpFunctionEnd
```

The error message is wrong, and misleading:  
```
error: line 8: Block(s) {7} are referenced but not defined in function 1
  %1 = OpFunction %void None %5
```
The value %7 _is_ defined in the function.  The problem is it's not a _label_ that is defined in the function.

### PR DESCRIPTION
**Pull request by JasperNV**
Fixes #1628.

### CODE REVIEW

File: source/val/validate_cfg.cpp [hunk 1]
```diff
@@ -155,6 +155,27 @@ spv_result_t ValidateBranchConditional(ValidationState_t& _,
...
+  // At least two operands (selector, default), any more than that are
+  // literal/target.
+
+  // target operands must be OpLabel
+  for (auto i = 2; i < num_operands; i += 2) {
+    const auto literal_ = inst->GetOperandAs<uint32_t>(i + 0);
+    const auto id = inst->GetOperandAs<uint32_t>(i + 1);
+
+    const auto target = _.FindDef(id);
+    if (!target || SpvOpLabel != target->opcode()) {
```
Comment (dneto0) [inline source/val/validate_cfg.cpp]: Hey @alan-baker are there other places in the validator where branches are checked.  All branches need to target another block in the same function.  For example.

Comment (alan-baker) [reply source/val/validate_cfg.cpp]: This look like the right spot. OpBranchConditional is checked in a similar spot. OpBranch seems to be unchecked currently.

Comment (JasperNV) [reply source/val/validate_cfg.cpp]: I could add support for OpBranch as well if wanted. That would likely be a separate issue / PR though.

Comment (alan-baker) [reply source/val/validate_cfg.cpp]: That's definitely an useful check if you don't mind adding it. Not a problem as a separate PR.

**Comment by JasperNV** [comment]
CI-shaderc-smoketest fail does not look like my fault.

**Comment by JasperNV** [comment]
Added similar check for OpBranch as well.

**Comment by JasperNV** [comment]
That test is trying to branch to a function. Before, the error message was "Block blah are referenced but not defined in function.". My code changes it to "Target is not OpLabel instruction".

It sounds like the cleanest fix is to just delete that test, then?

**Comment by alan-baker** [comment]
Its ok just to update the error. That is still an invalid test and the validator should catch it.

**Comment by JasperNV** [comment]
Pushed an update. AppVeyor tests are now passing.

**Comment by alan-baker** [comment]
@JasperNV PTAL at the bot failures. Formatting and unused variables it look like. I'll reach them when you push another commit.

**Comment by JasperNV** [comment]
Sorry about that (on Windows here, can't test the other builds). Pushed one more update.

**Pull request merged**